tutorials/intermediate/autoscale.py

```diff
@@ -0,0 +1,178 @@
+y = np.sinc(x)
+
+fig, ax = plt.subplots()
+ax.plot(x, y)
+fig.show()
+
+###############################################################################
+# Margins
+# -------
+# The default margin around the data limits is 5%:
+
+ax.margins()
+
+###############################################################################
+# The margins can be made larger using `~matplotlib.axes.Axes.margins`:
+
+fig, ax = plt.subplots()
+ax.plot(x, y)
+ax.margins(0.2, 0.2)
+fig.show()
+
+###############################################################################
+# In general, margins can be in the range (-0.5, ∞), where negative margins set
+# the axes limits to a fraction of the data range, and allow a zoom effect.
+# Using a single number for margins affects both axes, a single margin can be
+# customized using keyword arguments ``x`` or ``y``, but positional and keyword
+# interface cannot be combined.
+
+fig, ax = plt.subplots()
+ax.plot(x, y)
+ax.margins(y=-0.2)
+fig.show()
+
+###############################################################################
+# Sticky edges
+# ------------
+# Some :class:`.Artist` objects do not allow margins.  For example, setting
+# ``margin=0.2`` on ``plt.imshow`` does not affect the resulting plot.
+#
+
+xx, yy = np.meshgrid(x, x)
+zz = np.sinc(np.sqrt((xx - 1)**2 + (yy - 1)**2))
+
+fig, ax = plt.subplots(ncols=2, figsize=(12, 8))
+ax[0].imshow(zz)
+ax[0].set_title("margins unchanged")
+ax[1].imshow(zz)
+ax[1].margins(0.2)
+ax[1].set_title("margins(0.2)")
+fig.show()
+
+###############################################################################
+# This override of margins is determined by "sticky edges", a
+# property of :class:`.Artist` class that can suppress adding margins to axis
+# limits. The effect of sticky edges can be disabled on an Axes by changing
+# `~matplotlib.axes.Axes.use_sticky_edges`.
+# Artists have a property `.Artist.sticky_edges`, and the values of
+# sticky edges can be changed by writing to ``Artist.sticky_edges.x`` or
+# ``.Artist.sticky_edges.y``.
+#
+# The following example shows how overriding works and when it is needed.
+
+fig, ax = plt.subplots(ncols=3, figsize=(16, 10))
+ax[0].imshow(zz)
+ax[0].margins(0.2)
+ax[0].set_title("use_sticky_edges unchanged\nmargins(0.2)")
+ax[1].imshow(zz)
+ax[1].margins(0.2)
+ax[1].use_sticky_edges = False
+ax[1].set_title("use_sticky_edges=False\nmargins(0.2)")
+ax[2].imshow(zz)
+ax[2].margins(-0.2)
+ax[2].set_title("use_sticky_edges unchanged\nmargins(-0.2)")
+fig.show()
+
+###############################################################################
+# We can see that setting ``use_sticky_edges`` to False renders the image with
+# requested margins. Additionally, as is stated, sticky edges count for adding
+# a margin, therefore negative margin is not affected by its state, rendering
+# the third image within narrower limits and without changing the
+# `~matplotlib.axes.Axes.use_sticky_edges` property.
```

Comment:
```suggestion
# requested margins.
#
# While sticky edges don't increase the axis limits through extra margins,
# negative margins are still taken into accout. This can be seen in
# the reduced limits of the third image.
```